test/e2e/selectors/selectors_test.go

```
@@ -65,64 +63,80 @@ var _ = Describe("Field Selectors in Flight Control", Ordered, func() {
 	Context("Basic Functionality Tests", func() {
 		It("We can list devices and create resources", Label("77917"), func() {
 			By("Listing devices")
-			out, err := harness.CLI("get", "devices")
+			out, err := RunGetDevices(harness)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(out).To(ContainSubstring("NAME"))
 
 			By("create a complete fleet")
-			_, _ = harness.CLI("delete", "fleet")
-			out, err = harness.CLI("apply", "-f", util.GetTestExamplesYamlPath(fleetYAMLPath))
+			// Creating fleet
+			_, _ = ManageResource(harness, "delete", "fleet")
+			out, err = ManageResource(harness, "apply", fleetYAMLPath)
+			// Verifying fleet is valid
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(MatchRegexp(resourceCreated))
 			Expect(err).ToNot(HaveOccurred())
 			Expect(out).To(MatchRegexp(resourceCreated))
 
 			By("Get device info from the yaml")
-			deviceInfo = harness.GetDeviceByYaml(util.GetTestExamplesYamlPath(deviceYAMLPath))
+			deviceInfo = harness.GetDeviceByYaml(GetTestExamplesYamlPath(deviceYAMLPath))
 			deviceName = *deviceInfo.Metadata.Name
+			deviceBInfo = harness.GetDeviceByYaml(GetTestExamplesYamlPath(deviceBYAMLPath))
+			deviceBName = *deviceBInfo.Metadata.Name
 			Expect(deviceName).ToNot(BeEmpty())
+			Expect(deviceBName).ToNot(BeEmpty())
 
+			// deleting previous devices
 			_, _ = harness.CLI("delete", "device")
-			out, err = harness.CLI("apply", "-R", "-f", util.GetTestExamplesYamlPath(deviceYAMLPath))
-			time.Sleep(30 * time.Second) // to establish fleet before adding device to it
-			Eventually(func() error {
-				_, err := harness.CLI("get", "fleet")
-				return err
-			}, "30s", "1s").Should(BeNil(), "Timeout waiting for fleet to be ready")
+			// Creating new devices
+			_, _ = ManageResource(harness, "apply", deviceYAMLPath)
```

why do you ignore the error in this case?